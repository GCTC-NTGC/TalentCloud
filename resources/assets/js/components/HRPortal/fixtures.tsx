import { text, boolean, number } from "@storybook/addon-knobs";
import { JobAction } from "../../models/types";
import { JobStatus } from "../../models/lookupConstants";

export const jobActions: JobAction[] = [
  {
    activity: {
      count: 3,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "CS01",
    draft: {
      url: "https://google.ca",
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: true,
    preview: {
      url: "https://google.ca",
      text: "Preview Poster",
      title: "Preview what this Job Poster will look like.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Draft,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Front-end Developer",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 12,
    classification: "BK01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Published,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Cupcake Baker",
    userTime: 2,
  },
  {
    activity: {
      count: 0,
      new: {
        url: "https://google.ca",
        text: "New Activity",
        title: "Click here to see new activity on this posting.",
      },
    },
    applicants: 5,
    classification: "BS01",
    draft: {
      url: null,
      text: "View Draft",
      title: "View this draft Job Poster.",
    },
    managerTime: 8,
    owned: false,
    preview: {
      url: "https://google.ca",
      text: "View Poster",
      title: "View the published Job Poster.",
    },
    screeningPlan: {
      url: "https://google.ca",
      text: "View Screening Plan",
      title: "View the screening plan for this Job Poster.",
    },
    status: JobStatus.Complete,
    summary: {
      url: "https://google.ca",
      text: "Visit Job Summary",
      title: "View the Job Summary for this Job Poster.",
    },
    title: "Underwater Basket Weaver",
    userTime: 2,
  },
];

export const unclaimedJobs = [
  {
    title: "CS01 - Front-end Developer",
    createdAt: "2019-MAY-02",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: [],
  },
  {
    title: "AS02 - Executive Assisstant",
    createdAt: "2019-MAR-12",
    status: JobStatus.Draft,
    hiringManagers: ["Rebecca Appleby"],
    hrAdvisors: ["Rebecca Appleby", "Jack Little"],
  },
  {
    title: "ET03 - Business Analyst",
    createdAt: "2019-DEC-02",
    status: JobStatus.Review,
    hiringManagers: ["Robin Browne"],
    hrAdvisors: [],
  },
  {
    title: "FG05 - Long-term Care Specialist",
    createdAt: "2019-MAY-14",
    status: JobStatus.Draft,
    hiringManagers: ["Braeden McDoogal"],
    hrAdvisors: ["Caitlyn Summers", "Jack Little"],
  },
  {
    title: "CS03 - Digital Product Designer",
    createdAt: "2019-JUL-24",
    status: JobStatus.Review,
    hiringManagers: ["Amelie Lachance"],
    hrAdvisors: [],
  },
];
