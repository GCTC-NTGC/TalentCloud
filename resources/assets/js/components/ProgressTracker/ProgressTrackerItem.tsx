import React from "react";

interface ProgressTrackerItemProps {
  state: "active" | "complete" | "error" | "null";
  label: string;
  title: string;
  fontColor?: string;
}
const ProgressTrackerItem: React.FunctionComponent<
  ProgressTrackerItemProps
> = ({ label, title, state, fontColor }): React.ReactElement => {
  return (
    <div
      className="tracker-item"
      data-tc-tracker-state={state}
      data-c-alignment="base(left)"
      data-c-grid="middle"
      data-c-margin="top(half) right(normal) bottom(half) left(normal)"
      data-c-color={fontColor}
    >
      <div className="tracker-icon">
        {state === "active" && <i className="fas fa-arrow-down" />}
        {state === "complete" && <i className="fas fa-check" />}
        {state === "error" && <i className="fas fa-exclamation-triangle" />}
      </div>
      <div className="tracker-title">
        <span data-c-font-size="small">{label}</span>
        <span data-c-font-weight="bold">{title}</span>
      </div>
    </div>
  );
};

export default ProgressTrackerItem;
