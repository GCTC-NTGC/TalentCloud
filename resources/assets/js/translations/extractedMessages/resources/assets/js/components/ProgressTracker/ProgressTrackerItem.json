[{"id":"progressTracker.unreachableStep","description":"Tooltip informing user to complete previous steps, before step it can become clickable.","defaultMessage":"Must complete previous steps."}]
