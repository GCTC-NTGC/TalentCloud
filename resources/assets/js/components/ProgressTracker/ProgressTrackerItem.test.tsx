import React from "react";
import { shallow } from "enzyme";
import ProgressTrackerItem from "./ProgressTrackerItem";
import { items } from "./fixtures/progressItems";

describe("Progress Tracker", (): void => {
  it("should render ProgressTrackerItem correctly", (): void => {
    const wrapper = shallow(<ProgressTrackerItem {...items[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
