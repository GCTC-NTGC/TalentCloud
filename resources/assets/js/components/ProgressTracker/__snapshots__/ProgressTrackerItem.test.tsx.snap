// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Tracker should render ProgressTrackerItem correctly 1`] = `
<div
  className="tracker-item"
  data-c-alignment="base(left)"
  data-c-grid="middle"
  data-c-margin="top(half) right(normal) bottom(half) left(normal)"
  data-tc-tracker-state="active"
>
  <div
    className="tracker-icon"
  >
    <i
      className="fas fa-arrow-down"
    />
  </div>
  <div
    className="tracker-title"
  >
    <span
      data-c-font-size="small"
    >
      Step 01
    </span>
    <span
      data-c-font-weight="bold"
    >
      Job Info
    </span>
  </div>
</div>
`;
